### PR TITLE
Data-related updates, reduce CI matrix

### PR DESCRIPTION
**Pull request by paulromano**
The primary purpose of this PR is to update our documentation to mention our new website, https://openmc.mcs.anl.gov which hosts pregenerated HDF5 libraries. Since the openmc-dev/data repository is now hosting all our data conversion scripts, there's no need to have all the `openmc-convert-*` and `openmc-get-*` scripts here as users generally won't be worrying about running them, so I've removed them.

A few other random updates here:
- I've updated thermal names to recognize thermal scattering files from JEFF 3.3
- In an attempt to reduce our build times, I've reduced our build matrix to do special builds (with OpenMP, with MPI, with PHDF5, etc.) only with one version of Python. Fundamentally, there's nothing different from the Python side, so we're not really missing information by cutting out those builds.
- I added a script called `conda-install.sh` that installs OpenMC and all its prerequisites using miniconda and the Anaconda repos. The coolest thing is that is uses compilers from conda itself (gcc 7), so you can run it even on older linux distributions where the gcc version is insufficient. @pshriwise can vouch for its awesomeness.

### CODE REVIEW

**Comment by paulromano** [comment]
@Shimwell I also have a fix in here for #1161

File: scripts/conda-install.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+conda create -y -n $ENVNAME python=$PYTHON_VERSION
+conda activate $ENVNAME
+
+# Install recent version of gcc and CMake. OpenMC requires gcc 4.9+ and CMake
+# 3.3+, so if you are on an older distribution, it's possible that your version
+# is not sufficient.
+conda install -y gcc_linux-64 gxx_linux-64 gfortran_linux-64 cmake
+
+# Install OpenMC's Python dependencies
+conda install -y numpy scipy cython h5py pandas lxml matplotlib ipython pytest
```
Comment (liangjg) [inline scripts/conda-install.sh]: A minor issue here is `h5py` is a serial version by default, which will prevent any mpi-running of openmc in depletion or other similar capi-based simulations which will write out hdf5 results. So you may consider [installing parallel h5py using pip](http://docs.h5py.org/en/stable/build.html#building-against-parallel-hdf5):
``` bash
if [[ $WITH_MPI == yes ]]; then
    export HDF5_MPI="ON" 
    export HDF5_DIR=/usr/lib/x86_64-linux-gnu/hdf5/mpich
    pip install --no-binary=h5py h5py
fi
```

Comment (paulromano) [reply scripts/conda-install.sh]: Good point. I just fixed this, although slightly different than what you suggested (using parallel h5py from conda-forge).

File: scripts/conda-install.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+# INSTALLATION OPTIONS
+
+INSTALL_MINICONDA=yes                             # Install miniconda?
+INSTALL_IN_CONDA=yes                              # Install OpenMC executable/package directly in conda environment?
+INSTALL_EDITABLE=no                               # Install in editable mode (overrides INSTALL_IN_CONDA)
+INSTALL_PREFIX=$HOME/opt/openmc/$(date +%Y-%m-%d) # Where to install (if INSTALL_IN_CONDA and INSTALL_EDITABLE are disabled)
+WITH_MPI=yes                                      # Install with MPI support
+PYTHON_VERSION=3.7                                # What version of Python to use
+ENVNAME=openmc-py$PYTHON_VERSION                  # conda environment name
+
```
Comment (liangjg) [inline scripts/conda-install.sh]: How about adding an option to install nuclear cross section library, like `INSTALL_XSDATA`? Then scripts `tools/ci/download-xs.sh`  can be executed to download default data library, together with the appropriate environment variables activated? `INSTALL_NJOY` may also be considered.

File: scripts/conda-install.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+# Install MPICH for building with MPI support. Make sure to use the package
+# based on gcc7. For depletion, we also need the parallel version of h5py
+if [[ $WITH_MPI == yes ]]; then
+    conda install -y mpich
+    export CC=mpicc
+    export CXX=mpicxx
+    export FC=mpifort
+
+    conda install h5py=*=*mpich*
```
[the remark this answers is not in the document]
Comment (liangjg) [reply scripts/conda-install.sh]: I just tried to run this scripts in a clean system. It seems the `conda install -y h5py=*=*mpich*` will wipe the `CC`/`CXX`/`FC` variables, causing the error of not finding MPI in the following compiling step. So you may move the `export CC=...` lines below the `conda install -y h5py=*=*mpich*` .

**Comment by paulromano** [comment]
@liangjg I agree with your feedback on the conda-install script. Given that it will take me some time to fix it up and enhance it, I decided to separate it out from this PR. I'd prefer that these other fixes get in sooner. I'll put in another PR with that script once I feel it's ready. Thanks for reviewing!

**Comment by liangjg** [comment]
@paulromano Sure. The other fixes look good to me. I'd like to merge it before tonight if no one else gives more feedback.

**Pull request merged**